From 7f3ccedd019a250f424b6c0e035854c6539b905f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Mon, 6 Mar 2023 11:55:32 +0100
Subject: [PATCH 001/841] feat(st): add ST specific version

NOT_UPSTREAMABLE

Use a specific version for ST software to ease debug.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 1e02fd7be62225d780026afceee28f68b080b82a feat(st): add ST specific version]
Signed-off-by: Yann Gautier <yann.gautier@st.com>
Change-Id: I84311c915a9caaa4174259070c0de15c82718a5e
---
 plat/st/common/common.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/st/common/common.mk b/plat/st/common/common.mk
index 22998cb6e..bc31c6c59 100644
--- a/plat/st/common/common.mk
+++ b/plat/st/common/common.mk
@@ -10,6 +10,11 @@ STM32MP_EARLY_CONSOLE		?=	0
 STM32MP_RECONFIGURE_CONSOLE	?=	0
 STM32MP_UART_BAUDRATE		?=	115200
 
+# Add specific ST version
+ST_VERSION 			:=	r1.0
+ST_GIT_SHA1			:=	$(shell git rev-parse --short=8 HEAD 2>/dev/null)
+VERSION_STRING			:=	v${VERSION_MAJOR}.${VERSION_MINOR}-${PLAT}-${ST_VERSION}(${BUILD_TYPE}):${BUILD_STRING}(${ST_GIT_SHA1})
+
 TRUSTED_BOARD_BOOT		?=	0
 STM32MP_USE_EXTERNAL_HEAP	?=	0
 
-- 
2.47.2

